engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbUserTaskState.java

```diff
@@ -56,9 +72,34 @@ public void update(final UserTaskRecord userTask) {
   }
 
   @Override
-  public void delete(final long userTaskKey) {
-    this.userTaskKey.wrapLong(userTaskKey);
-    userTasksColumnFamily.deleteExisting(this.userTaskKey);
+  public void update(final long key, final Consumer<UserTaskRecord> modifier) {
```

Comment:
:thinking: I'm wondering how we'll perform the tenant authorization check with this method. I guess it will be similar to the `update` method, where we'll first need to `get` the `UserTask` while providing authorization information.

Reply:
When we get to the additional lifecycle actions (like the `update` I think you're referencing here?), we'll probably have to refactor this to be an "internal" method and do a tenant check with a get before we do the internal update, yes. Is that what you're referring to?

Reply:
Yes, I'm referring to what you mentioned in [the comment above](https://github.com/camunda/zeebe/pull/15433#discussion_r1417310182). It's fine if we leave it like this, I was just thinking out loud.

Reply:
Alright, I actually had similar thoughts when I did the cancelation 👍 We'll tackle it when we get there 🙂